src/components/_channel/ChannelWithVideos/ChannelWithVideos.tsx

```diff
@@ -50,16 +50,27 @@ export const ChannelWithVideos: FC<ChannelWithVideosProps> = ({ channelId }) =>
 
   const targetItemsCount = videosPerRow * videoRows
   const displayedVideos = (videos || []).slice(0, targetItemsCount)
-  const placeholderItems = videosLoading ? Array.from({ length: targetItemsCount }, () => ({ id: undefined })) : []
+  const placeholderItems = useMemo(
+    () => (videosLoading ? Array.from({ length: targetItemsCount }, () => ({ id: undefined })) : []),
+    [targetItemsCount, videosLoading]
+  )
 
-  const gridContent = (
-    <>
-      {[...displayedVideos, ...placeholderItems].map((video, idx) => (
-        <VideoTile id={video.id} key={`channels-with-videos-${idx}`} showChannel />
-      ))}
-    </>
+  const gridContent = useMemo(
+    () => (
+      <>
+        {[...displayedVideos, ...placeholderItems].map((video, idx) => (
+          <VideoTile id={video.id} key={`channels-with-videos-${idx}`} showChannel />
+        ))}
+      </>
+    ),
+    [displayedVideos, placeholderItems]
```

Comment:
Not your change, but we actually don't need to wrap this in `Fragment`, right?
```suggestion
  const gridContent = useMemo(
    () =>
      [...displayedVideos, ...placeholderItems].map((video, idx) => (
        <VideoTile id={video.id} key={`channels-with-videos-${idx}`} showChannel />
      )),
    [displayedVideos, placeholderItems]
  )
```